ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Keystore/Types.hs

```diff
@@ -0,0 +1,101 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Ariadne.Wallet.Cardano.Kernel.Keystore.Types
+       ( StorageDecodingError (..)
+       , InternalStorage (..)
+       , Version (..)
+       , isFilePath
+       , isWallets
+       , isVersion
+       , getInternalStorageKeys
+       ) where
+
+import Control.Lens.TH (makeLenses)
+import Data.Aeson
+  (FromJSON, ToJSON, Value(..), object, parseJSON, toJSON, withObject, (.:), (.=))
+import Data.Aeson.Types (Parser)
+import Data.Default (Default, def)
+import Data.Map (Map)
+import Data.Version (Version(..), makeVersion, parseVersion, showVersion)
+import Formatting (bprint, build)
+import Text.ParserCombinators.ReadP (readP_to_S)
+
+import Pos.Binary.Class (decodeFull', serialize')
+import Pos.Core (AddressHash)
+import Pos.Crypto (EncryptedSecretKey(..), PublicKey)
+
+import Ariadne.Wallet.Cardano.Kernel.Keystore.Util (esksToMap)
+
+import qualified Data.Map as Map (elems, empty)
+import qualified Data.Text.Buildable
+import qualified Data.Vector as V
+import qualified Serokell.Util.Base16 as S (decode, encode)
+
+-- | InternalStorage data containing wallet keys that gets saved to file
+data InternalStorage
+  = InternalStorage
+  { _isWallets  :: !(Map (AddressHash PublicKey) EncryptedSecretKey)
+  , _isVersion  :: !Version
```

Comment:
We don't need this field in the data type. We don't use it anywhere. We should put `version` into serialized JSON value and check it during deserialization. But we shouldn't carry it further.